### PR TITLE
[SPARK-30119][WEBUI]Support pagination for streaming tab

### PR DESCRIPTION
**Pull request by iRakson**
### What changes were proposed in this pull request?
#28747 reverted #28439 due to some flaky test case. This PR fixes the flaky test and adds pagination support.

### Why are the changes needed?
To support pagination for streaming tab

### Does this PR introduce _any_ user-facing change?
Yes, Now streaming tab tables will be paginated.

### How was this patch tested?
Manually.

### CODE REVIEW

**Comment by iRakson** [comment]
cc @sarutak Kindly take a look.

**Comment by iRakson** [comment]
retest this please

**Comment by iRakson** [comment]
cc @srowen I forgot to cc you

File: streaming/src/test/scala/org/apache/spark/streaming/UISeleniumSuite.scala [hunk 1]
```diff
@@ -125,24 +125,37 @@ class UISeleniumSuite
 
         // Check batch tables
         val h4Text = findAll(cssSelector("h4")).map(_.text).toSeq
-        h4Text.exists(_.matches("Active Batches \\(\\d+\\)")) should be (true)
         h4Text.exists(_.matches("Completed Batches \\(last \\d+ out of \\d+\\)")) should be (true)
```
Comment (sarutak) [inline streaming/src/test/scala/org/apache/spark/streaming/UISeleniumSuite.scala:128]: Is this still flaky isn't it? What if it reaches this line before at least one batch finishes?

Comment (iRakson) [reply streaming/src/test/scala/org/apache/spark/streaming/UISeleniumSuite.scala:128]: It should be fine now. In 2nd attempt it should get at least 1 complete batch. As second attempt is after 500 ms and batches are of 100 ms only.

**Comment by iRakson** [comment]
@sarutak gentle ping

**Comment by sarutak** [comment]
Seems OK to me.

**Review by sarutak — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by iRakson** [comment]
Thank You @sarutak @srowen

**Pull request closed without merging**